src/components/points/DistanceComponent.jsx

```diff
@@ -22,18 +22,18 @@ function DistanceComponent({ id, latitude, longitude, classes, proximityToUnlock
 
   function unlockThisPoint(id) {
     setListOfUnlocked([...listOfUnlocked, ...[id]]);
-    const currentLocalStorage = localStorage.getItem("unlocked-experiences");
+    const currentLocalStorage = localStorage.getItem(`unlocked-experiences-${selectedRoute?.id}`);
```

Comment:
Put the storage key into a variable and use it in all (3) places where you need it:

```suggestion
    const storageKey = `unlocked-experiences-${selectedRoute?.id}`
    const currentLocalStorage = localStorage.getItem(storageKey);
```